aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -629,6 +630,37 @@ at::Tensor preprocess_mask(
 
   return attn_mask;
 }
+// FlashAttentionV2 requires that head dimension be a multiple of 8
+// This was previously done within the kernel, however
+// This causes the kernel to maybe alias query, key, value
+// So instead we pad the head_dimensions to be a multiple of 8 in the composite
+// region
+template <int alignment_size, bool slice>
+at::Tensor pad_last_dim(const at::Tensor& attn_bias) {
```

Comment:
Why is the head dimension the last dim? I though that's the embedding dimension.

Reply:
I'm probably just mixing up terminologies here.

Reply:
so by the time you have called sdpa, you have taken your `embed_dim` and split it into `num_heads * head_dim`